agent/agent.go

```diff
@@ -318,3 +324,24 @@ func initializeStateManager(cfg *config.Config, taskEngine engine.TaskEngine, cl
 	}
 	return stateManager, nil
 }
+
+// initializeENIWatcher wraps up the setup for creating the ENI Watcher
+func initializeENIWatcher(ctx context.Context) {
+	log.Debug("Setting up ENI Watcher")
+	// Create UDev Monitor
+	udevMonitor, err := udev.NewMonitor()
+	if err != nil {
+		log.Errorf("Error creating udev monitor: %v", err)
+		return
+	}
+	// Create Watcher
+	watcher := eniWatcher.New(ctx, netlinkWrapper.NetLinkClient{}, udevMonitor)
+
+	err = watcher.Init()
+	if err != nil {
+		log.Errorf("Error initializing ENI Watcher: %v", err)
+		watcher.Stop()
+		return
+	}
+	go watcher.Start()
```

Comment:
I think that this method should return a handle to the `watcher` object so that the invoking method has a chance to call something like `defer watcher.Stop()` if/when needed.